### PR TITLE
refactor(core): Cleaned up registration function

### PR DESCRIPTION
**Pull request by ice0**
Removed all unnecessary "magic"

### CODE REVIEW

File: synfig-core/src/synfig/valuenode_registry.cpp [hunk 1]
```diff
@@ -52,23 +52,13 @@ String
 ValueNodeRegistry::localize_name(const String &local_name)
 	{ return _(local_name.c_str()); }
 
-void
+bool
```
Comment (rodolforg) [inline synfig-core/src/synfig/valuenode_registry.cpp:55]: why `bool` if it always will return true?

File: synfig-core/src/synfig/valuenode_registry.h [hunk 2]
```diff
@@ -35,38 +35,13 @@
...
-			return _local_name;												\
-		}																	\
-																			\
-		const ReleaseVersion Register_##klass::release_version = _version;	\
-		const char* Register_##klass::name = _name;							\
-		const char* Register_##klass::local_name = _local_name;				\
-	}
+/// get_name & get_local_name methods.
+#define REGISTER_VALUENODE(klass, _version, _name, _local_name) \
+static bool result_##klass = synfig::ValueNodeRegistry::register_node_type(_name, \
```
Comment (rodolforg) [inline synfig-core/src/synfig/valuenode_registry.h:40]: Don't you think indentation would help realize next lines are part of `#define`?

File: synfig-core/src/synfig/valuenode_registry.cpp (hunk 1 above)
Comment (ice0) [reply synfig-core/src/synfig/valuenode_registry.cpp:55]: To call the auto-registration function "automatically", we create a static variable. In this case, we need some sort of return value.

So all the "magic" of auto-registration is to make something like this:
```
static bool result = register_me("class_name");
```
You can't do this on void function.

File: synfig-core/src/synfig/valuenode_registry.h (hunk 2 above)
Comment (ice0) [reply synfig-core/src/synfig/valuenode_registry.h:40]: How do you propose to indent them?

Comment (ice0) [reply synfig-core/src/synfig/valuenode_registry.h:40]: @rodolforg 
Done.

**Comment by hesham-essam** [comment]
This is much better, and it works with msvc. The old trick is such a head scratcher, I still don't know what does that even mean
```
template<do_register&> struct register_ref {}; // do_register is a variable not a type!
```

**Comment by ice0** [comment]
> This is much better, and it works with msvc. The old trick is such a head scratcher, I still don't know what does that even mean
> 
> ```
> template<do_register&> struct register_ref {}; // do_register is a variable not a type!
> ```

Same for me. It looks like a hack for MSVC compiler, but I am not sure about it.

Anyway, if you check/review the PR and don't see any issues, please approve it, just to show that someone else has checked it too.

**Review by hesham-essam — APPROVED** [review]

**Review by morevnaproject — APPROVED** [review]

**Pull request merged**